### ISSUE TITLE
Bug report: Length of nil value

### ISSUE THREAD

**Issue by zepinglee** [zepinglee/citeproc-lua#77]
Hi, I get the same error with v0.6.3 and with the `lualatex` engine, when trying to show two citations in two different chapters: 

```tex
\documentclass{book}

\usepackage[style=elsevier-harvard]{citation-style-language}
\usepackage{hyperref}

\addbibresource{references}

\begin{document}

\chapter{Introduction}\label{introduction}

See \citet{knuth84} for additional discussion of literate programming.

\chapter{Summary}\label{summary}

In summary, this book has no content whatsoever \cite{knuth84}.

\chapter*{References}\label{references}

\printbibliography[heading=none]

\end{document}
```

giving this error:

```
...dist/scripts/citation-style-language/citeproc-engine.lua:755: attempt to get length of a nil value (field '?')
stack traceback:
        ...dist/scripts/citation-style-language/citeproc-engine.lua:755: in method 'get_preceding_cite_item'
        ...dist/scripts/citation-style-language/citeproc-engine.lua:708: in method 'set_cite_item_position'
        ...dist/scripts/citation-style-language/citeproc-engine.lua:647: in method 'get_tainted_citation_ids'
        ...dist/scripts/citation-style-language/citeproc-engine.lua:585: in method 'process_citation'
        ...ist/scripts/citation-style-language/citeproc-manager.lua:502: in method 'cite'
        [\directlua]:1: in main chunk.
\lua_now:e #1->\__lua_now:n {#1}
```

_Originally posted by @julienleroux5 in https://github.com/zepinglee/citeproc-lua/issues/75#issuecomment-2334357327_

**Comment by zepinglee** [comment]
@julienleroux5 This bug is already fixed in [510769c](https://github.com/zepinglee/citeproc-lua/commit/510769c11e0e92856c067b9166f4815a4f14504a). I've published to CTAN and it should be available soon.

**Comment by julienleroux5** [comment]
Cool, working great, thanks so much!
